### PR TITLE
Add plugin build

### PR DESCRIPTION
**Pull request by willgittoes-dd**
Creates a shared plugin, as suitable for linking into nginx-opentracing.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -271,6 +274,27 @@ jobs:
     <<: *package_nginx_module_base
     <<: *image_ubuntu_16_04
 
+  # Builds the dynamic plugin as well as the shared lib.
```
Comment (cgilmour) [inline .circleci/config.yml]: The CMAKE_ARGS seem to say a different thing from this comment.

File: .circleci/config.yml [hunk 2]
```diff
@@ -282,7 +306,7 @@ jobs:
           name: Install Datadog OpenTracing nginx module
           command: |
             CODENAME=$(lsb_release -c | cut -f2) # eg "bionic", "xenial"
-            apt install -y ./packages/datadog-opentracing-nginx*${CODENAME}*.deb
+            apt install -y ./artifacts/datadog-opentracing-nginx*${CODENAME}*.deb
             ldconfig
```
Comment (cgilmour) [inline .circleci/config.yml]: It'd be nice if ldconfig wasn't necessary after installation.
Can we do things to make it not necessary?

File: .circleci/config.yml [hunk 3]
```diff
@@ -297,41 +321,58 @@ jobs:
             ./nginx_integration_test.sh
 
 
+_workflow_filters:
+  _version_tag: &version_tag
+    # Allows semver or "test" tags, with any suffix.
+    # eg v1.2.3 v0.1.1 v1.2.3-ayy-lmao test test-ayy-lmao
```
Comment (cgilmour) [inline .circleci/config.yml]: 🤣

File: .circleci/config.yml [hunk 4]
```diff
@@ -297,41 +321,58 @@ jobs:
...
-            tags:
-              only: /.*/
+          <<: *run_always
       - integration_test_nginx_ubuntu_18_04:
           requires:
             - package_nginx_module_ubuntu_18_04
-          filters:
-            tags:
-              only: /.*/
+          <<: *run_always
```
Comment (cgilmour) [inline .circleci/config.yml]: This is now officially more complicated than the code it builds.

File: .circleci/config.yml (hunk 3 above)
Comment (willgittoes-dd) [reply .circleci/config.yml]: 👽

File: .circleci/config.yml (hunk 2 above)
Comment (willgittoes-dd) [reply .circleci/config.yml]: Known issue, I've just not yet added lfconfig to package postinst (eg https://people.canonical.com/~cjwatson/ubuntu-policy/policy.html/ch-sharedlibs.html )

File: .circleci/config.yml (hunk 4 above)
Comment (willgittoes-dd) [reply .circleci/config.yml]: What I wanted to do was have two workflows, one called "build_test", the other "build_test_release". First one runs always, second one runs on version tags (or test tags, so we can test it without making a release). 

But circleci doesn't allow per-workflow triggers that aren't cron jobs. So I had to do this instead :'(

File: .circleci/config.yml (hunk 1 above)
Comment (willgittoes-dd) [reply .circleci/config.yml]: "I changed my mind" (and now I updated the comment)

File: .circleci/config.yml [hunk 5]
```diff
@@ -152,26 +151,26 @@ _jobs:
             # Version number "v1.2.3" -> "1.2.3", otherwise leave unmodified.
             DD_CPP_VERSION=$(git describe --tags --always  | sed -n -e 's/v\([0-9]*\.[0-9]*\.[0-9]*\)/\1/' -e 'p')
             CODENAME=$(lsb_release -c | cut -f2) # eg "bionic", "xenial"
-            mkdir -p $PACKAGE_DIR
+            mkdir -p $ARTIFACT_DIR
             echo "Building .deb with dd-opentracing-cpp version ${DD_CPP_VERSION}-${CODENAME}"
             # TODO(willgittoes-dd): Add changelog to package.
             fpm -s dir -t deb -n datadog-opentracing \
-              -C $BUILD_DIR -p $PACKAGE_DIR/datadog-opentracing_VERSION_ARCH.deb \
+              -C $BUILD_DIR -p $ARTIFACT_DIR/datadog-opentracing_VERSION_ARCH.deb \
```
[the remark this answers is not in the document]
Comment (willgittoes-dd) [reply .circleci/config.yml]: Yes, they're templated. It just uses the version that I pass in! Bit silly innit?

**Pull request merged**